panther_manager/include/panther_manager/moving_average.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef PANTHER_MANAGER_MOVING_AVERAGE_HPP_
+#define PANTHER_MANAGER_MOVING_AVERAGE_HPP_
+
+namespace panther_manager
+{
+
+template <typename T>
+class MovingAverage
+{
+public:
+  MovingAverage(int window_size = 5) : window_size_(window_size), sum_(T(0)) {}
+
+  void add_value(T value)
```

Comment:
```suggestion
  void roll(T value)
```
add value implies extending the underlying array. Roll is similar to how they named it in [pandas](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.rolling.html)